spec/factories.rb

```diff
@@ -30,6 +30,17 @@
   factory :project do |p|
     p.name 'Test Project'
     p.start_date { Time.current }
+
+    trait :with_tag do
+      after(:build) { |object| object.tag_group = create(:tag_group) }
```

Comment:
Can't you just use an association here? https://github.com/thoughtbot/factory_girl/blob/master/GETTING_STARTED.md#associations